plugins/package-managers/spm/src/main/kotlin/SpmModel.kt

```diff
@@ -90,35 +90,35 @@ data class AppDependency(
         val revision: String? = null,
         val branch: String? = null
     )
+}
 
-    fun toPackage(): Package {
-        val id = Identifier(
-            type = PACKAGE_TYPE,
-            namespace = "",
-            name = getCanonicalName(repositoryUrl),
-            version = state?.run {
-                when {
-                    !version.isNullOrBlank() -> version
-                    !revision.isNullOrBlank() -> "revision-$revision"
-                    !branch.isNullOrBlank() -> "branch-$branch"
-                    else -> ""
-                }
-            }.orEmpty()
-        )
-
-        val vcsInfoFromUrl = VcsHost.parseUrl(repositoryUrl)
-        val vcsInfo = if (vcsInfoFromUrl.revision.isBlank() && state != null) {
+internal fun AppDependency.toPackage(): Package {
```

Comment:
Ok, now this cannot be a property anymore, also no an extension property, as access to `state` is required, which comes from the parent class.